app/scripts/controllers/preferences.js

```diff
@@ -406,16 +407,17 @@ class PreferencesController {
    * end of the list. The current list is modified and returned as a promise.
    *
    * @param {string} _url The rpc url to add to the frequentRpcList.
+   * @param {bool} remove Remove selected url
    * @returns {Promise<array>} The updated frequentRpcList.
    *
    */
-  addToFrequentRpcList (_url) {
+  addToFrequentRpcList (_url, remove = false) {
     const rpcList = this.getFrequentRpcList()
     const index = rpcList.findIndex((element) => { return element === _url })
     if (index !== -1) {
       rpcList.splice(index, 1)
     }
-    if (_url !== 'http://localhost:8545') {
+    if (!remove && _url !== 'http://localhost:8545') {
       rpcList.push(_url)
     }
     if (rpcList.length > 3) {
```

Comment:
Since you've added the remove option, we can probably remove this list limiting block.